src/particle/corsika/corsika7id.py

```diff
@@ -0,0 +1,168 @@
+# Copyright (c) 2018-2022, Eduardo Rodrigues and Henry Schreiner.
+#
+# Distributed under the 3-clause BSD license, see accompanying file LICENSE
+# or https://github.com/scikit-hep/particle for details.
+
+"""
+Class representing a Corsika7 ID.
+
+Note
+----
+Corsika8 uses Geant3 Particle IDs.
+"""
+
+
+from __future__ import annotations
+
+import csv
+from typing import TypeVar
+
+from .. import data
+from ..exceptions import MatchingIDNotFound
+from ..pdgid import PDGID
+
+Self = TypeVar("Self", bound="Corsika7ID")
+
+
+with data.basepath.joinpath("pdgid_to_corsika7id.csv").open() as _f:
+    _bimap = {
+        int(v["CORSIKA7ID"]): int(v["PDGID"])
+        for v in csv.DictReader(line for line in _f if not line.startswith("#"))
+    }
+
+# Some Corsika ID's are not really particles
```

Comment:
Both versions 7 and 8? Asking since you mention above that Corsika8 uses Geant3 IDs. Worth being specific if relevant.

Reply:
Actually I'm not really sure, since I have (not yet) worked with Corsika8. I'll specify 7.